### PR TITLE
Move ipu precision flag check to IPUPrecisionPlugin init

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?

Part of  #11449 follow up item 1

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/plugins/precision/ipu.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ class IPUPrecisionPlugin(PrecisionPlugin):
     """Precision plugin for IPU integration."""
 
     def __init__(self, precision: int) -> None:
+        if precision not in (16, 32):
+            raise MisconfigurationException(f"`Trainer(accelerator='ipu', precision={precision!r})` is not supported.")
```
Comment (ananthsub) [inline pytorch_lightning/plugins/precision/ipu.py:34]: The error message should include what *is supported* so users can quickly resolve the issue without digging into the source code or docs.

Comment (ananthsub) [inline pytorch_lightning/plugins/precision/ipu.py:34]: The error message should include what *is supported* so users can quickly resolve the issue without digging into the source code or docs.
```suggestion
        supported_precision_values = (16, 32)
        if precision not in supported_precision_values:
            raise MisconfigurationException(f"`Trainer(accelerator='ipu', precision={precision!r})` is not supported. `precision` must be one of: {supported_precision_values}")
```

File: pytorch_lightning/plugins/precision/ipu.py [hunk 2]
```diff
@@ -30,6 +30,8 @@ class IPUPrecisionPlugin(PrecisionPlugin):
     """Precision plugin for IPU integration."""
```
Comment (ananthsub) [inline pytorch_lightning/plugins/precision/ipu.py:30]: docstring should include that it raises a misconfiguration error if the precision value is not supported

Comment (kaushikb11) [reply pytorch_lightning/plugins/precision/ipu.py]: Updated!

**Review by kaushikb11 — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

File: pytorch_lightning/plugins/precision/ipu.py [hunk 3]
```diff
@@ -27,9 +27,20 @@
...
+
+    Raises:
+        MisconfigurationException:
+            If the precision is neither 16 nor 32.
+    """
 
     def __init__(self, precision: int) -> None:
+        supported_precision_values = (16, 32)
+        if precision not in supported_precision_values:
+            raise MisconfigurationException(
```
Comment (carmocca) [inline pytorch_lightning/plugins/precision/ipu.py:40]: #10918 

```suggestion
            raise ValueError(
```

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 4]
```diff
@@ -660,11 +660,6 @@ def _check_and_init_precision(self) -> PrecisionPlugin:
     def _validate_precision_choice(self) -> None:
         """Validate the combination of choices for precision, AMP type, and accelerator."""
         # TODO: change exception type to ImpactableConfigurationException
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:662]: Remove this TODO?

```suggestion
```

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 4 above)
Comment (four4fish) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:662]: This is still a valid TODO i think, @justusschock has a issue about it

Comment (four4fish) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:662]: Oh! The one you mentioned above :)

File: pytorch_lightning/plugins/precision/ipu.py (hunk 3 above)
Comment (four4fish) [reply pytorch_lightning/plugins/precision/ipu.py:40]: Should we add a ImpactableConfigurationException? or use valueError?

Comment (carmocca) [reply pytorch_lightning/plugins/precision/ipu.py:40]: I think good old `ValueError` is good for now :)

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 4 above)
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:662]: Yes, but you moved the exception but not the TODO

Still, I think `ValueError` is fine.

Comment (four4fish) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:662]: Sounds good :)

File: pytorch_lightning/plugins/precision/ipu.py [hunk 5]
```diff
@@ -27,9 +27,20 @@
...
+        MisconfigurationException:
+            If the precision is neither 16 nor 32.
+    """
 
     def __init__(self, precision: int) -> None:
+        supported_precision_values = (16, 32)
+        if precision not in supported_precision_values:
+            raise ValueError(
+                f"`Trainer(accelerator='ipu', precision={precision!r})` is not supported."
+                f" `precision` must be one of: {supported_precision_values}."
```
Comment (awaelchli) [inline pytorch_lightning/plugins/precision/ipu.py:42]: So the reason we had the check in the accelerator connector is because the error message is not accurate when instantiating the plugin (or a subclass of it) outside the Trainer. So yeah, not ideal to move it here :)

File: pytorch_lightning/plugins/precision/ipu.py [hunk 6]
```diff
@@ -27,9 +27,20 @@
 
 
 class IPUPrecisionPlugin(PrecisionPlugin):
-    """Precision plugin for IPU integration."""
+    """Precision plugin for IPU integration.
+
+    Raises:
+        MisconfigurationException:
```
Comment (awaelchli) [inline pytorch_lightning/plugins/precision/ipu.py:33]:
```suggestion
        ValueError:
```

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/precision/ipu.py (hunk 5 above)
Comment (four4fish) [reply pytorch_lightning/plugins/precision/ipu.py:42]: @awaelchli do you mean that user may init the precision plugin by themselves? Are you suggesting keep this in accl_connt?

Comment (awaelchli) [reply pytorch_lightning/plugins/precision/ipu.py:42]: > do you mean that user may init the precision plugin by themselves? 

in general yes, that's what the plugins are meant to be used for. For the IPU precision specifically, nobody will. Still, the error message making a reference to the Trainer is misleading.
